mkosi/archive.py

```diff
@@ -102,3 +102,19 @@ def make_cpio(src: Path, dst: Path, files: Optional[Iterable[Path]] = None) -> N
         # Make sure tar uses user/group information from the root directory instead of the host.
         options=finalize_passwd_mounts(dst),
     )
+
+
+def normalize_mtime(root: Path, source_date_epoch: Optional[str]) -> None:
```

Comment:
I don't think this function fits here. We should do the environment variable parsing in the MkosiState constructor and then expose a property on MkosiState with the parsed value. And then just have a function normalize_mtime() in the `__init__` module that takes the  parsed mtime as an input.